AdoptAWidget: ColorFiltered

Muskan4010 opened the pull request:
AdoptAWidget: ColorFiltered

<!-- Provide an explanation of the change -->

<!-- Specify what this pull request does -->
This pull request is:

- [x] A code sample
- [x] More references
- [ ] More explanation

<!-- Add the AdoptAWidget issue that is assigned to you: -->
closes #69491

Muskan4010 committed:
Updated API docs ColorFilter

Muskan4010 committed:
Updated API docs ColorFilter

scheglov committed:
Ignore several import_of_legacy_library_into_null_safe (#69904)

* Ignore import_of_legacy_library_into_null_safe instead of un-migrating.

liyuqian committed:
Reland migration (#69996)

This reverts ae0a9cb560ff3b334ab0cc23980374792888b2d1 and therefore relands #69629.

Additionally, `flutter update-packages --force-upgrade` has been run to update `pubspec.yaml` to resolve the analyze test failures.

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

Muskan4010 committed:
references updated

jiahaog committed:
[integration_test] Add a setter for defaultTestTimeout (#70073)

* [integration_test] Add a setter for defaultTestTimeout

* type annotation

* More type annotations

* Fix doc

xster committed:
move cupertino_icon template to 1.0.1 for null safety (#69997)

mit-mit committed:
Roll engine and fix pubspecs that do not have a Dart SDK constraint (#70078)

scheglov committed:
Ignore several import_of_legacy_library_into_null_safe (#70107)

HansMuller committed:
AppBar draws its defaults from theme.colorScheme (#69251)

chunhtai committed:
Fix crash when widgetspan does not produce a semantics node in render… (#69919)

* Fix crash when widgetspan does not produce a semantics node in render paragraph

* fix comments

* fixed corner case, added test, refactored semantics widget

* addressing comment

scheglov committed:
Ignore import_of_legacy_library_into_null_safe (#70115)

Commit:
[flutter_tools] remove workaround for caching sound dill (#70014)

engine-flutter-autoroll committed:
Roll Engine from 3cf292226986 to 40b8f8227b9a (7 revisions) (#70117)

Commit:
Migrate Flutter Gallery test to null safety (#70116)

Commit:
[null-safety] migrate hello_world to null safety (#70112)

* [null-safety] migrate flutter gallery to null safety

Commit:
[flutter_tools] restore pub caching functionality on run/test (#70056)

Commit:
Bump Gallery version (#70130)

Commit:
Revert "[flutter_tools] restore pub caching functionality on run/test (#70056)" (#70132)

This reverts commit 6e5845f0253f5574c0bd059808650b2d419260bb.

yjbanov committed:
default to unsound nullability for web (#70120)

xu-baolin committed:
Fix the PopupMenuButton offset bug (#69383)

engine-flutter-autoroll committed:
7f07491bc Roll Skia from cfe647c02fb4 to ee0ce9858cbc (1 revision) (flutter/engine#22399) (#70123)

Piinks committed:
Remove old todo (#70131)

Commit:
[flutter_tools] always run pub with prebuilt applicaitons on drive (#70136)

engine-flutter-autoroll committed:
Roll Engine from 7f07491bcafb to 014da89eb015 (3 revisions) (#70135)

redsolver committed:
AdoptAWidget: Table (#69488) (#69588)

* Added Dartpad sample for Table widget

* Changed template to stateless_widget_scaffold

* Removed trailing character

gspencergoog committed:
Remove references to `Window`, and switch usages to `PlatformDispatcher` or `SingletonFlutterWindow` (#69617)

* Remove references to dart:ui.Window, and point usages to PlatformDispatcher or SingletonFlutterWindow, as appropriate

* remove new test platform dispatchers

* Amend documentation

johnpryan commented on packages/flutter/lib/src/widgets/color_filter.dart:
````diff
@@ -12,6 +12,43 @@ import 'framework.dart';
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return  Column(
+///     children:[
+///       ColorFiltered(
+///         colorFilter: ColorFilter.mode(
+///           Colors.red,
+///           BlendMode.modulate,
+///         ),
+///         child:Image.network('https://flutter.github.io/assets-for-api-docs/assets/widgets/owl-2.jpg'),
+///       ),
+///       ColorFiltered(
+///         colorFilter: ColorFilter.mode(
+///           Colors.grey,
+///           BlendMode.saturation,
+///         ),
+///         child:Image.network('https://flutter.github.io/assets-for-api-docs/assets/widgets/owl.jpg'),
+///       ),
+///     ],
+///   );
````
Since can overflow, I would consider wrapping this in a SingleChildScrollView:

```suggestion
///     return SingleChildScrollView(
///       child: Column(
///         children: [
///           ColorFiltered(
///             colorFilter: ColorFilter.mode(
///               Colors.red,
///               BlendMode.modulate,
///             ),
///             child: Image.network(
///                 'https://flutter.github.io/assets-for-api-docs/assets/widgets/owl-2.jpg'),
///           ),
///           ColorFiltered(
///             colorFilter: ColorFilter.mode(
///               Colors.grey,
///               BlendMode.saturation,
///             ),
///             child: Image.network(
///                 'https://flutter.github.io/assets-for-api-docs/assets/widgets/owl.jpg'),
///           ),
///         ],
///       ),
///     );
```

johnpryan commented on packages/flutter/lib/src/widgets/color_filter.dart:
```diff
@@ -12,6 +12,43 @@ import 'framework.dart';
+///
+/// See Also:
+///
+/// *[Blendmode], blending a source image with destination image.
+/// *[Image],the class in the [dart:ui](https://api.flutter.dev/flutter/dart-ui/dart-ui-library.html) library.
+/// *[Colors], constants which represent Material design's [color palette](https://material.io/design/color/).
+/// *Cookbook:[Display images from the internet](https://flutter.dev/docs/cookbook/images/network-image)
```
A few fixes: (formatting, BlendMode, Material Design)

```suggestion
///  * [BlendMode], for describing how to blend a source image with destination image.
///  * [Image], the class in the [dart:ui](https://api.flutter.dev/flutter/dart-ui/dart-ui-library.html) library.
///  * [Colors], constants which represent Material Design's [color palette](https://material.io/design/color/).
///  * Cookbook: [Display images from the internet](https://flutter.dev/docs/cookbook/images/network-image)
```

johnpryan reviewed (changes requested):
Looks great overall!

Commit:
[flutter_tools] restore pub caching functionality (#70133)

Necessary to work around #70013 (this looks like it will be fixed, but should protect us in the future)

Restores some notion of skipping pub, with differences from the prior implementation:

Off by default, and only used by the flutter command automatic pub get
A single check parameter. Uses the framework version cached per-project instead of a timestamp of the tool, to avoid cases where downloading a prebuilt framework would cause this check to no longer work.

Commit:
[null-safety] implement null-safe autodetection for the web (#70126)

Fixes #69416
Fixes #70121

engine-flutter-autoroll committed:
Roll Engine from 014da89eb015 to d642a939c34b (2 revisions) (#70140)

Commit:
Revert "[flutter_tools] restore pub caching functionality (#70133)" (#70144)

This reverts commit ff953e4f83d0e9bc5a5468a7e910a7b1506a2ca5.

Commit:
[flutter_tools] remove most globals from asset system and remove Cache manipulation in unit tests (#70011)

darrenaustin committed:
Fix for the ListTile horizontalTitleGap calculation introduced with PR #64222. (#70149)

engine-flutter-autoroll committed:
Roll Engine from d642a939c34b to e50ac2c55240 (2 revisions) (#70158)

Commit:
[flutter_tools] always use dart to run test script. (#70146)

Commit:
Revert "[flutter_tools] always use dart to run test script. (#70146)" (#70175)

This reverts commit a90880a8c32905dfa6f204348ad6a9e6d404d282.

engine-flutter-autoroll committed:
Roll Engine from e50ac2c55240 to 4e0f7ab788bd (2 revisions) (#70163)

Commit:
[flutter_tools] dont use autodetect enum for web (#70189)

Commit:
[flutter_tools] remove experiment tracking analytics for null safety (#70183)

HansMuller committed:
Revert "AppBar draws its defaults from theme.colorScheme (#69251)" (#70212)

Commit:
[flutter_tools] reland: cache pub invocations (#70180)

Disabled caching of pub invocations on flutter drive.

goderbauer committed:
Assert for RenderFlex intrinsics if using baseline alignment (#70139)

engine-flutter-autoroll committed:
Roll Engine from 4e0f7ab788bd to caf678d30fc0 (10 revisions) (#70231)

* 03e750d6f Roll Fuchsia Linux SDK from g6EuxMthn... to DzZi2gPbF... (flutter/engine#22417)

* 80e9a3fc2 Roll Skia from 84d503b21322 to 5b8598952931 (7 revisions) (flutter/engine#22418)

* 80cc0fae7 Roll Skia from 5b8598952931 to 02dd0ed8ce5e (1 revision) (flutter/engine#22419)

* d14c4a742 Roll Skia from 02dd0ed8ce5e to fb5850f41043 (4 revisions) (flutter/engine#22420)

* af185be87 Roll Skia from fb5850f41043 to 008d63e23dab (6 revisions) (flutter/engine#22421)

* 1a13dac52 Simplify API for scheduling Skia object deletions (flutter/engine#22409)

* 49299c37d Roll Skia from 008d63e23dab to 267826c86552 (4 revisions) (flutter/engine#22422)

* 25e08298a Roll Fuchsia Mac SDK from w10eytxvc... to e-4Jm-yWa... (flutter/engine#22423)

* 76e615859 Roll Skia from 267826c86552 to 88e8bb2fe2d5 (3 revisions) (flutter/engine#22424)

* caf678d30 Move common graphics utils to //flutter/common/graphics (flutter/engine#22320)

engine-flutter-autoroll committed:
5f660ce78 Roll ANGLE to latest version (flutter/engine#22177) (#70237)

Commit:
[null-safety] opt driver tests into null safety (#70245)

keyonghan committed:
mark catalina ios tasks as flaky (#70255)

engine-flutter-autoroll committed:
Roll Engine from 5f660ce78474 to c5cbbb000f2c (5 revisions) (#70251)

engine-flutter-autoroll committed:
3de24322a Roll Skia from 1fe2b80dc782 to 24c18526a564 (1 revision) (flutter/engine#22432) (#70264)

xster committed:
Update cupertino and material translations (#70157)

engine-flutter-autoroll committed:
81f219c59 [Android Text Input] Make the editing state listenable and allow batch edits  (flutter/engine#21534) (#70266)

johnsonmh committed:
[Material] Add support for customizing active + disabled state color for selection controls. (#68831)

* Add new MaterialStateProperty param to selection controls

Chinmay-KB committed:
AdoptAWidget: WillPopScope (#69563)

* Update WillPopScope API Docs

* Update WillPopScope API Docs

* Update WillPopScope API Docs template fix

* Refactoring dartpad sample

* Refactoring dartpad sample

* Remove trailing spaces

* Minor build fix

* Remove space

* Change the dartpad sample and other minor changes

* Change the dartpad example

* Remove trailing character

* refactor code to dart preamble

* adding uncleared variable

* adding uncleared variable

* Implemented requested changes

* Implemented requested changes

* empty commit

* minor changes

Co-authored-by: Chinmay Kabi <chinmaykabi@gmail.com>

obitodarky committed:
AdoptAWidget: Stepper (#70092)

* interactive dartpad sample added

* cleaner code

* removed extra spaces

* minor spaces added

* removed null returns

sanjoli63 committed:
Update ReorderableListView API docs (#69650)

* Update ReorderableListView API docs

* fixed dart fmt

* added dartpad instead of code sample

* fixed parenthesis error

* Update packages/flutter/lib/src/material/reorderable_list.dart

Co-authored-by: Amit Patil <54329870+Amitpatil215@users.noreply.github.com>

* using reference List.generate instead of generate

* request changes are complete

* space eroors are fixed

* Update ReorderableListView API docs

* fixed dart fmt

* added dartpad instead of code sample

* fixed parenthesis error

* using reference List.generate instead of generate

* Update packages/flutter/lib/src/material/reorderable_list.dart

Co-authored-by: Amit Patil <54329870+Amitpatil215@users.noreply.github.com>

* request changes are complete

* space eroors are fixed

* request changes are completes

Co-authored-by: Amit Patil <54329870+Amitpatil215@users.noreply.github.com>
Co-authored-by: John Ryan <ryjohn@google.com>

Commit:
[flutter_tools] migrate .packages to package_config, partial (#70200)

OmarEmaraDev committed:
Correct GtkKeyHelper key codes for the Meta key (#69576)

Piinks committed:
Default Keyboard ScrollActions with PrimaryScrollController (#69795)

Commit:
[flutter_tools] improve hash performance in build_system (#70065)

Commit:
[flutter_tools] make getBuildInfo async (#70320)

jpnurmi committed:
Fix Editable(Text) shortcuts to respect read-only on desktop (#69891)

Commit:
Move web integration tool tests to web.shard (#70226)

* Move web integration tool tests to web.shard

Web integration tool tests depend on DDC changes in SDK. This change
moves them to a separate shard and subshard so CI bot configurations
can run them separately.

In particular, with will allow running those tests on dart CI flutter
HHH web bot instead of a non-web one, allowing early detection and easy
classification of issues caused by SDK changes as VM- or Web related.

* Enabled verbose mode for flaky web_tool_tests

* Split out the test changes to be commited first

engine-flutter-autoroll committed:
Roll Engine from 81f219c59c62 to c5c7e83b9e53 (3 revisions) (#70317)

keyonghan committed:
Revert "mark catalina ios tasks as flaky (#70255)" (#70328)

engine-flutter-autoroll committed:
Roll Engine from c5c7e83b9e53 to d97a81c88983 (6 revisions) (#70333)

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

Muskan4010 committed:
SingleChildScrollView added and references updated

zanderso committed:
Revert "Roll Engine from c5c7e83b9e53 to d97a81c88983 (6 revisions) (#70333)" (#70388)

This reverts commit 15d2d8a8750c2733871f7f8ae558c268efd32fc0.

zanderso committed:
Revert "Default Keyboard ScrollActions with PrimaryScrollController (#69795)" (#70391)

This reverts commit 1f0df54553bab852026a2801ce3921a2ba493fab.

goderbauer committed:
Intrinsic Width fixes for RenderParagraph (#70236)

Commit:
[flutter_tools] make most integration tests null safe (#70334)

gspencergoog committed:
Fix Platform channel errors in web tests (#70252)

Commit:
Future can return null (#70324)

HansMuller committed:
Update OutlinedButton default outline geometry to be backwards compatible (#70393)

gspencergoog committed:
Revert usages of the binding's platformDispatcher to use window instead (#70319)

This reverts usages of the binding's platformDispatcher to use window again temporarily, because there isn't a TestPlatformDispatcher yet, and so some tests were failing because they mocked the TestWindow to return certain things (locales) that were returning the real values instead of the test values.

Once I've created a TestPlatformDispatcher to allow fake data to be passed to it, we can go back to using the platformDispatcher in all of these places

tvolkert committed:
Ignore DismissIntent when barrier is not dismissible (#70156)

jason-simmons committed:
Set the mixWithOthers option in the Gallery video demo (#70396)

This allows the two videos in the demo to play simultaneously
(matching the behavior from before the recent upgrade to the
video player plugin)

liyuqian committed:
Add SkiaPerfPoint and FlutterEngineMetricPoint (#70153)

Unit tests that cover their translations are also added.

Commit:
[flutter_tools] allow using flutter test for testing the tool too (#69911)

goderbauer committed:
Fix return type of ContextAction.invoke (#70397)

goderbauer committed:
Properly initialize RestorationManager in the TestBinding (#70398)

Commit:
Revert "Add SkiaPerfPoint and FlutterEngineMetricPoint (#70153)" (#70407)

This reverts commit a0ec4d67e9d9312ad88c1ba1bb11483edb8fb15a.

darrenaustin committed:
Migrate flutter_localizations to null safety. (#68645)

Migrate flutter_localizations to null safety.

jmagman committed:
Add -miphoneos-version-min=8.0 to App framework stub (#70405)

Commit:
[flutter_tools] remove branch migration and standardize constructor style for version interface (#70058)

gspencergoog committed:
Allow propagation to ancestor actions if actions lower down are disabled (#70404)

Change the semantics of Actions.invoke so that if the action it finds is disabled, then it keeps looking for an ancestor Actions widgets that has a matching intent where the action is not disabled.

gspencergoog committed:
Add casts to widgets_test.dart to fix the engine roll. (#70423)

engine-flutter-autoroll committed:
Roll Engine from c5c7e83b9e53 to dddb532b5c7b (40 revisions) (#70429)

jiahaog committed:
[integration_test] Add a `run` method for proper reporting of test results (#70075)

Commit:
Revert "[integration_test] Add a `run` method for proper reporting of test results (#70075)" (#70466)

This reverts commit af5eb3b987e1749692ff226be18b087006908e4e.

Commit:
[flutter_tools] use initially parsed package config for language version, sound mode determination (#70323)

goderbauer committed:
Enable test after engine fix (#70469)

Commit:
[devicelab] LUCI builder flag (#70464)

engine-flutter-autoroll committed:
Roll Engine from dddb532b5c7b to bbcf19ad143c (6 revisions) (#70477)

* 03ff8a409 TBR: remove the extra values which was overshadowing the test result (flutter/engine#22483)

* e9b6a428f Roll Skia from a06b63c56ecd to 8ead30d51c86 (1 revision) (flutter/engine#22487)

* ee700513e Roll Fuchsia Linux SDK from fULjPqtx9... to B4PaMsNWM... (flutter/engine#22490)

* 4cbb684ef Roll Dart SDK from 3e502e0c7e04 to 41893ff76b0f (1 revision) (flutter/engine#22491)

* 38b6c22d5 PlatformViewIOS CreateExternalViewEmbedder refactor (flutter/engine#22353)

* bbcf19ad1 Roll Skia from 8ead30d51c86 to 011218edb590 (4 revisions) (flutter/engine#22493)

liyuqian committed:
e2e perf tests don't have CPU/GPU/memory metrics (#70439)

Address comment in https://github.com/flutter/flutter/pull/66386/files#r522671103

Commit:
[flutter_tools] remove automatic doctor from flutter create (#70475)

Piinks committed:
Revert "Actually consume top padding in bottomsheet if scrollcontrolled (#66257)" (#70392)

Commit:
[flutter_tools] remove unused JSON schema (#70480)

Commit:
remove testUsingContext (#70482)

Hixie committed:
Simplify the flutter_web_plugins plugin registration API. (#70337)

In principle this is backwards compatible; I just merged all the classes into one and deprecated everything that became redundant as a result.

engine-flutter-autoroll committed:
Roll Engine from bbcf19ad143c to 63a6fe3cf9ef (3 revisions) (#70490)

Commit:
Disable failing test due to SDK issue (#70485)

* Disable failing test due to SDK issue

* Disable failing debugger and hot reload windows tests

Commit:
Revert "Simplify the flutter_web_plugins plugin registration API. (#70337)" (#70505)

This reverts commit 8749d95521c710c5c1869927f90b22457614f6de.

gspencergoog committed:
Remove some unnecessary casts, now that changes have rolled from the engine (#70491)

engine-flutter-autoroll committed:
Roll Engine from 63a6fe3cf9ef to e06f10638598 (3 revisions) (#70494)

liyuqian committed:
Reland "Add SkiaPerfPoint and FlutterEngineMetricPoint (#70153)" (#70419)

This reverts commit 2a5aa29442271b8018425dc7dd2389aff638e5de.

Additionally, fix a few NNBD issues.

Fixes https://github.com/flutter/flutter/issues/70408

Commit:
Revert "[devicelab] LUCI builder flag (#70464)" (#70496)

creativecreatorormaybenot committed:
Add stackTrace to AsyncSnapshot (#69507)

jiahaog committed:
[integration_test] Reland add a `run` method for proper reporting of test results (#70513)

* [integration_test] Reland add a `run` method for proper reporting of test results

* Changes to path resolving from previous PR

Commit:
[flutter_tools] stop unit test from writing real file (#70511)

engine-flutter-autoroll committed:
Roll Engine from e06f10638598 to 3e77d854a7cb (18 revisions) (#70554)

tiya-google committed:
Add bottom to search bar (#68794)

dnfield committed:
Increase timeouts on XCUITests (#70631)

engine-flutter-autoroll committed:
Roll Engine from 3e77d854a7cb to fae92ed57427 (6 revisions) (#70624)

jason-simmons committed:
Ignore text selection boxes when assembling semantics for placeholder runs (#70487)

IsmailAbdirahman committed:
AdoptAWidget: NotificationListener (#69524)

* Added notificationListner sample code

* Fixed spacing

* fixed spacing

* Moved the variables into the Build method

* fixed typos

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

* Update packages/flutter/lib/src/widgets/notification_listener.dart

Co-authored-by: John Ryan <ryjohn@google.com>

Hixie committed:
More lints (#70500)

Commit:
[flutter_tools] Add bot configuration to run web_tool_tests for linux, mac, and windows (#70412)

* Move web integration tool tests to web.shard

Web integration tool tests depend on DDC changes in SDK. This change
moves them to a separate shard and subshard so CI bot configurations
can run them separately.

In particular, with will allow running those tests on dart CI flutter
HHH web bot instead of a non-web one, allowing early detection and easy
classification of issues caused by SDK changes as VM- or Web related.

* Enabled verbose mode for flaky web_tool_tests

* fixed local engine crash with --enable-asserts

* Disable failing tests, fix local engine with asserts

* Removed mac web_tool_tests bot

* Addressed CR comments

* Add disabled mac web_tool_tests and disable win web_tool_tests

Piinks committed:
Re-land 'Default Keyboard ScrollActions with PrimaryScrollController' (#70401)

christopherfujino committed:
skip roll_dev_integration_test (#70653)

jmagman committed:
Remove unused host_app_editable_cocoapods template files (#70518)

Commit:
Add liblzma as an explicit dependancy on Linux (#70617)

gspencergoog committed:
Remove last references to Window in docs. (#70637)

This removes the last couple of doc references to the Window class in the framework, so that I can re-land the engine side change that removes the Window class (flutter/engine#22505).

jmagman committed:
Always build x86 mac apps (#70649)

Commit:
[flutter_tools] remove globals from features (#70515)

engine-flutter-autoroll committed:
Roll Engine from fae92ed57427 to f0e80100f740 (8 revisions) (#70657)

flar committed:
Revert "Re-land 'Default Keyboard ScrollActions with PrimaryScrollController' (#70401)" (#70675)

This reverts commit 9b56241eef3a81bfe4e1ff72cc453a7693861944.

xu-baolin committed:
Add a [valid] property of [MouseTrackerAnnotation] indicates the annotation states. (#69866)

engine-flutter-autoroll committed:
Roll Engine from f0e80100f740 to 635df897603a (3 revisions) (#70665)

engine-flutter-autoroll committed:
Roll Engine from 635df897603a to 8d93ea0d06e9 (4 revisions) (#70696)

jmagman committed:
Move Flutter.framework to build directory instead of ios/Flutter (#70224)

sidrao2006 committed:
AdoptAWidget - Progress indicator (#69498)

* Add examples for LinearProgressIndicator

* Add examples for CircularProgressIndicator

* Change name Linear  -> Circular

* Remove unnecessary changes

* Add images to docs

* Fix semi-colon bugs

* Fix trailing spaces

* Fix more semicolon bugs

* Remove imports in docs

* Remove error causing line breaks

* Add ticker providers

* Add more ticker providers

* Remove assets and combine samples

* Removed 6 samples

* Remove unnecessary animation

Co-authored-by: John Ryan <ryjohn@google.com>

* Remove unnecessary animation

Co-authored-by: John Ryan <ryjohn@google.com>

* Use material theme

Co-authored-by: John Ryan <ryjohn@google.com>

* Apply suggestions from code review

Co-authored-by: John Ryan <ryjohn@google.com>

Commit:
[flutter_tools] remove unused/deprecated asset flags (#70509)

engine-flutter-autoroll committed:
Roll Engine from 8d93ea0d06e9 to f3cc39a5d646 (2 revisions) (#70704)

Commit:
[devicelab] LUCI builder flag (#70702)

Commit:
[flutter_tools] reduce build bundle API (#70514)

jmagman committed:
Add extra Flutter settings to ios_app_with_extensions Podfile (#70719)

jmagman committed:
Turn off bitcode in ios_app_with_extensions to match normal Flutter projects (#70724)

bkonyi committed:
Support legacy behavior for --host-vmservice-port and --observatory-port with DDS (#70336)

Implements the following:
  - If both --host-vmservice-port and --dds-port are specified, use the current behavior
  - If only --host-vmservice-port is specified and dds is enabled, use that for the dds port
  - If only --dds-port is specified, use that for the dds port

Fixes https://github.com/flutter/flutter/issues/70332

bkonyi committed:
Roll package:dds to 1.5.1 and add isCompleted guards around completers in base/dds.dart (#70712)

flar committed:
preserve detailFiles entry in TaskResult JSON (#70668)

* preserve detailFiles entry in TaskResult JSON

yash1200 committed:
Updated tests in material/bottom_navigation_bar_test.dart (#66670)

johnsonmh committed:
[Material] Add method to get dark mode overlay color without needing BuildContext (#70669)

engine-flutter-autoroll committed:
Roll Engine from f3cc39a5d646 to 94e217bcf609 (7 revisions) (#70721)

xu-baolin committed:
Deprecated unused property [RectangularSliderTrackShape.disabledThumbGapWidth] (#65246)

Commit:
[gen-l10n] NNBD generated code (#70415)

* NNBD gen-l10n generated code

jmagman committed:
Allow any iOS app to be added to an existing host app (#70647)

dnfield committed:
Revert "Allow any iOS app to be added to an existing host app (#70647)" (#70739)

This reverts commit abdd65fc29d8d97cb2e057c2b50688d6e5baa88b.

dnfield committed:
Revert "Revert "Allow any iOS app to be added to an existing host app (#70647)" (#70739)" (#70740)

This reverts commit aab9a76ee4cd97199e9d21118a0a3cfbda80b899.

engine-flutter-autoroll committed:
Roll Engine from 94e217bcf609 to 35a0b9fe6827 (5 revisions) (#70736)

guidezpl committed:
Adaptive icons (#69119)

* refactor update_icons

* fix trailing space

* address feedback

* add platform adaptive icons

* fix merge conflict

* Update dartdoc

* Address feedback

* Specify types

* Add tests

* fix indentation

* Remove trailing space

* Remove isCupertino static bool

stuartmorgan-g committed:
Clean up Windows plugin template (#70403)

Removes a comment that is left over from early iterations of the
template.

gspencergoog commented on packages/flutter/lib/src/widgets/color_filter.dart:
```diff
@@ -12,6 +12,47 @@ import 'framework.dart';
+///
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// The sample shows how to filter images with selected color
```
```suggestion
/// This sample shows how to filter images with a color
```

gspencergoog commented on packages/flutter/lib/src/widgets/color_filter.dart:
```diff
@@ -12,6 +12,47 @@ import 'framework.dart';
+/// {@tool dartpad --template=stateless_widget_scaffold}
+///
+/// The sample shows how to filter images with selected color
+/// and blending them with different modes
```
```suggestion
/// and blend them using the [BlendMode.modulate] and [BlendMode.saturation] modes.
```

nero-angela committed:
[Tabs] Fix tab indicator flies off issue (#65463)

gspencergoog commented on packages/flutter/lib/src/widgets/color_filter.dart:
```diff
@@ -12,6 +12,47 @@ import 'framework.dart';
 /// Applies a [ColorFilter] to its child.
 ///
```
Could you also please add some more text here about the class itself? Something like:

"Causes the child to be modified using a color to tint the resulting rendering. Different blending modes can modify how the color is applied, and to which parts of the image."

Just to describe what the class is for.

christopherfujino committed:
fix and re-enable integration test (#70666)

chinmoy12c committed:
added none property in a DismissDirection (#68987)

jmagman committed:
Force regeneration of old Podfile (#70735)

Commit:
[flutter_tools] add support for dart defines to flutter test (#70791)

Commit:
[devicelab] update hot reload benchmark to be more representative (#70643)

update hot reload benchmark to be more representative

Commit:
[flutter_tools] use frontend_server for web test compilation (#70714)

jmagman committed:
Stop extra framework copy during build ios-framework (#70790)

Commit:
Revert "Stop extra framework copy during build ios-framework (#70790)" (#70802)

This reverts commit a9a35d10a49bead426c91e487370402a0cf1793a.

Commit:
[flutter_tools] refactor shared memory filesystem logic (#70795)

Commit:
[flutter_tools] update dependencies (#70797)

Muskan4010 committed:
change in grammatical errors,added class description

Muskan4010 committed:
change in grammatical errors,added class description

Muskan4010 committed:
Description of widget added

Muskan4010 committed:
Description of widget added

Muskan4010 committed:
decription of widget

Muskan4010 committed:
decription of widget

Muskan4010 committed:
spaces edited

Muskan4010 committed:
spaces edited

jason-simmons committed:
Fix type cast null safety in MethodChannel._invokeMethod (#70809)

dnfield committed:
remove ephemeral files (#70818)

jmagman committed:
Run module tests on try builders when flutter_tools changes (#70804)

chinmoy12c committed:
Added enableFeedback property PopupMenuButton (#69890)

Commit:
[flutter_tools] skip ck restart on all platforms (#70853)

Commit:
[flutter_tools] remove globals from flutter web platform (#70863)

Commit:
[flutter_tools] wire up alternative invalidation strategy to features (#70865)

jmagman committed:
Detect ARM ffi CocoaPods error, suggest gem install (#70801)

Commit:
Revert "[flutter_tools] wire up alternative invalidation strategy to features (#70865)" (#70884)

This reverts commit ef4741540bb52b4ccc20ada8db7410557ae5b53c.

dnfield committed:
skip flaky test (#70881)

Skip flaky iOS XCUITest

yjbanov committed:
Clean-up docs for the --web-renderer option (#70880)

JuYeong0413 committed:
Update FlexibleSpaceBar dartpad sample (#70382)

guidezpl committed:
Fix typo (#70834)

engine-flutter-autoroll committed:
Roll Engine from 35a0b9fe6827 to a0da844845f5 (20 revisions) (#70871)

Commit:
[flutter_tools] run web unit tests in sound null safety (#70799)

Piinks committed:
Re-land 'Default Keyboard ScrollActions with PrimaryScrollController' (#70893)

royarg02 committed:
[flutter_tools] Make flutter upgrade --verify-only display framework version differences instead of commit hashes (#69420)

gspencergoog committed:
Remove the `nullOk` parameter from `Navigator.of` and add `Navigator.maybeOf` (#70726)

Adds Navigator.maybeOf to replace calling Navigator.of(context, nullOk: true), and removes the nullOk parameter. Also changes Navigator.of to return a non-nullable value, and removes many (120!) instances of the ! operator, reducing the possible places where a null dereference could occur.

perclasson committed:
[Material] Decoration for DataTable is not used inside PaginatedDataTable (#70603)

engine-flutter-autoroll committed:
Roll Engine from a0da844845f5 to 23a8e027dbbe (5 revisions) (#70904)

jpnurmi committed:
[flutter_tools] deploy version.json asset on Linux (#69920)

jmagman committed:
Remove deprecated 'flutter build aot' (#70898)

Commit:
flutter_tools: refactor `CreateCommand`. (#70874)

jcollins-g committed:
Update dartdoc to 0.37.0 (#70882)

Commit:
[flutter_tools] display message for current null safety mode (#70718)

jmagman committed:
Uncomment build_ios_framework_module_test tmp cleanup (#70963)

Commit:
[flutter_tools] share bootstrap module between run and test (#70903)

Commit:
[devicelab] dont use set modified for hot reload bench (#70965)

Commit:
Migrate template to Gradle 6.7 and AGP 4.1.0 (#70808)

Commit:
refactor (#70959)

Commit:
only use code single path for verification of target file existence (#70962)

Hixie committed:
Minor documentation fixes (#70501)

darrenaustin committed:
Material Date Picker code restructure (#70708)

Date Picker restructuring. Moved files out of 'pickers' and merged several of the non-public implementation files into date_picker.dart.

Commit:
[flutter_tools] disable SWR optimization on dev (#70912)

Piinks committed:
Fix pointer scroll for nested NeverScrollables (#70953)

goderbauer committed:
Fixes Intrinsics for RenderParagraph and RenderWrap (#70656)

Commit:
Wait for Android plugin to load before configuring plugin dependency (#70970)

perclasson committed:
[Material] Add selection control themes (#70311)

Commit:
[flutter_tools] fix port leak in flutter_driver (#70999)

chulwoo-park committed:
Remove unnecessary argument from Row and Column constructor comments (#71041)

Muskan4010 committed:
Merge branch 'colorfilter' of https://github.com/Muskan4010/flutter into colorfilter

jmagman commented:
This PR has 999 open files and cc'd 54 developers.  Please fix your branch against master and recreate this PR.  See https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request for help.
Thanks!
